### ISSUE TITLE
@RepositoryRestResource Jackson MismatchedInputException when POSTing JPA entity link relationship

### ISSUE THREAD

**Issue by NickGrantONS** [spring-projects/spring-data-rest#2043]
In Spring Boot 2.5.2, something is broken with Jackson, when using `@RepositoryRestResource` to POST a new entity into the DB.

In SB v2.4.8 the following works OK:

```java
import lombok.Data;

@Data
@Entity
public class CollectionExercise {
  @Id private UUID id;

  @ManyToOne private Survey survey;
}

@Data
@Entity
public class Survey {
  @Id private UUID id;

  @OneToMany(mappedBy = "survey")
  private List<CollectionExercise> collectionExercises;
}


@RepositoryRestResource
public interface CollectionExerciseRepository
    extends PagingAndSortingRepository<CollectionExercise, UUID> {}
```

Then, I can do this: `POST http://localhost:8080/collectionExercises` with the following body:
```json
{
  "id": "246b3010-6373-421d-9e9f-ed1366f17385",
  "survey": "surveys/5af5b469-0cd3-45e9-9692-5253917e3f62"
}
```

In SB v2.5.2 I get the following exception:

```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `bla.blala.Survey` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('http://localhost:8080/surveys/5af5b469-0cd3-45e9-9692-5253917e3f62')
```

Downgrading to SB 2.4.8 works fine.

**Comment by odrotbohm** [comment]
Would you mind trying the Spring Data snapshots? We fixed #2033 that might have interfered with the deserializer setup.

**Comment by NickGrantONS** [comment]
> Would you mind trying the Spring Data snapshots? We fixed #2033 that might have interfered with the deserializer setup.

Sure no problem

**Comment by NickGrantONS** [comment]
Yep this is fixed in 3.5.2-SNAPSHOT.

Example POM extract:

```xml
  <repositories>
    <repository>
      <id>spring-snapshots</id>
      <url>https://repo.spring.io/snapshot</url>
      <snapshots><enabled>true</enabled></snapshots>
    </repository>
  </repositories>

  ...

    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-data-rest</artifactId>
      <exclusions>
        <exclusion>
          <groupId>org.springframework.data</groupId>
          <artifactId>spring-data-rest-webmvc</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>org.springframework.data</groupId>
      <artifactId>spring-data-rest-webmvc</artifactId>
      <version>3.5.3-SNAPSHOT</version>
    </dependency>
```
